packages/block-library/src/image/image.js

```diff
@@ -80,6 +80,8 @@ export default function Image( {
 	onUploadError,
 	containerRef,
 } ) {
+	const captionRef = useRef();
+	const prevUrl = useRef();
```

Comment:
There's a dedicated hook for this sort of thing in the `compose` package: [`usePrevious`](https://github.com/WordPress/gutenberg/blob/master/packages/compose/src/hooks/use-previous/README.md).